src/OpenTelemetry.Api/Context/RuntimeContext.cs

```diff
@@ -27,10 +28,66 @@ public static class RuntimeContext
     {
         private static readonly ConcurrentDictionary<string, object> Slots = new();
 
+        private static Type contextSlotType;
+
+        private static RuntimeContextSlotFactory runtimeContextSlotFactory;
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode", Justification = "User-defined RuntimeContextSlotFactory that relies on Reflection is not timmer safe.")]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL3050:RequiresDynamicCode", Justification = "User-defined RuntimeContextSlotFactory that relies on Reflection is not AoT safe.")]
```

Comment:
```suggestion
        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode", Justification = "The AsyncLocalRuntimeContextSlot type is handled without using Reflection..")]
        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL3050:RequiresDynamicCode", Justification = "The AsyncLocalRuntimeContextSlot type is handled without using Reflection.")]
```

Reply:
For code reviewers:
These 2 attributes are suppressing the below warnings:
```
C:\repos\opentelemetry-dotnet\src\OpenTelemetry.Api\Context\RuntimeContext.cs(39): Trim analysis warning IL2026: OpenTelemetry.Context.RuntimeContext..cctor(): Using member 'OpenTelemetry.Context.RuntimeContext.ContextSlotType.set' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. Setting a custom ContextSlotType requires using Reflection to initialize the context slot. [C:\repos\opentelemetry-dotnet\test\OpenTelemetry.AotCompatibility.TestApp\OpenTelemetry.AotCompatibility.TestApp.csproj]
C:\repos\opentelemetry-dotnet\src\OpenTelemetry.Api\Context\RuntimeContext.cs(39): AOT analysis warning IL3050: OpenTelemetry.Context.RuntimeContext..cctor(): Using member 'OpenTelemetry.Context.RuntimeContext.ContextSlotType.set' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. Setting a custom ContextSlotType requires using MakeGenericType to initialize the context slot. The native code for the generic type might not be available at runtime. [C:\repos\opentelemetry-dotnet\test\OpenTelemetry.AotCompatibility.TestApp\OpenTelemetry.AotCompatibility.TestApp.csproj]
```